app/code/community/Bolt/Boltpay/Model/BoltOrder.php

```diff
@@ -307,6 +326,41 @@ function ($item) use ($quote, &$calculatedTotal) {
         return $this->getCorrectedTotal($calculatedTotal, $cartSubmissionData);
     }
 
+    /**
```

Comment:
@oldDriver 
This is an unrelated refactor and should be placed in a separate PR.  Additionally, though you created it, you do not use it in the code.

Please make a separate PR for this change.  Keep this PR strictly to adding the new discount type.